types/views/views.go

```diff
@@ -21,6 +24,80 @@ func unmarshalSliceFromJSON[T any](b []byte, x *[]T) error {
 	return json.Unmarshal(b, x)
 }
 
+// ByteSlice is a read-only accessor for types that are backed by a []byte.
+type ByteSlice[T ~[]byte] struct {
+	// ж is the underlying mutable value, named with a hard-to-type
+	// character that looks pointy like a pointer.
+	// It is named distinctively to make you think of how dangerous it is to escape
+	// to callers. You must not let callers be able to mutate it.
+	ж T
+}
+
+// ByteSliceOf returns a ByteSlice for the provided slice.
+func ByteSliceOf[T ~[]byte](x T) ByteSlice[T] {
+	return ByteSlice[T]{x}
+}
+
+// Len returns the length of the slice.
+func (v ByteSlice[T]) Len() int {
+	return len(v.ж)
+}
+
+// IsNil reports whether the underlying slice is nil.
+func (v ByteSlice[T]) IsNil() bool {
+	return v.ж == nil
+}
+
+// Mem returns a read-only view of the underlying slice.
+func (v ByteSlice[T]) Mem() mem.RO {
+	return mem.B(v.ж)
+}
+
+// Equal reports whether the underlying slice is equal to b.
+func (v ByteSlice[T]) Equal(b T) bool {
+	return bytes.Equal(v.ж, b)
+}
+
+// EqualView reports whether the underlying slice is equal to b.
+func (v ByteSlice[T]) EqualView(b ByteSlice[T]) bool {
+	return bytes.Equal(v.ж, b.ж)
+}
+
+// AsSlice returns a deep-copy of the underlying slice.
```

Comment:
"deep"? It's a byte slice by definition.

Reply:
Copy paste fail, will fix